src/coreclr/jit/targetloongarch64.cpp

```diff
@@ -24,4 +24,216 @@ const regNumber fltArgRegs [] = {REG_F0, REG_F1, REG_F2, REG_F3, REG_F4, REG_F5,
 const regMaskTP fltArgMasks[] = {RBM_F0, RBM_F1, RBM_F2, RBM_F3, RBM_F4, RBM_F5, RBM_F6, RBM_F7 };
 // clang-format on
 
+//-----------------------------------------------------------------------------
+// LoongArch64Classifier:
+//   Construct a new instance of the LoongArch64 ABI classifier.
+//
+// Parameters:
+//   info - Info about the method being classified.
+//
+LoongArch64Classifier::LoongArch64Classifier(const ClassifierInfo& info)
+    : m_info(info)
+    , m_intRegs(intArgRegs, ArrLen(intArgRegs))
+    , m_floatRegs(fltArgRegs, ArrLen(fltArgRegs))
+{
+}
+
+//-----------------------------------------------------------------------------
+// Classify:
+//   Classify a parameter for the LoongArch64 ABI.
+//
+// Parameters:
+//   comp           - Compiler instance
+//   type           - The type of the parameter
+//   structLayout   - The layout of the struct. Expected to be non-null if
+//                    varTypeIsStruct(type) is true.
+//   wellKnownParam - Well known type of the parameter (if it may affect its ABI classification)
+//
+// Returns:
+//   Classification information for the parameter.
+//
+ABIPassingInformation LoongArch64Classifier::Classify(Compiler*    comp,
+                                                      var_types    type,
+                                                      ClassLayout* structLayout,
+                                                      WellKnownArg wellKnownParam)
+{
```

Comment:
Hi, @jakobbotsch 
Can we modify this 
```cplusplus
void LoongArch64Classifier::Classify(Compiler*    comp,
                                                      var_types    type,
                                                      ClassLayout* structLayout,
                                                      WellKnownArg wellKnownParam
                                                      ABIPassingInformation &abi)
{
```
to avoid construct function ?

Reply:
Why? I don't really see good reason to change it. `ABIPassingInformation` is just 2 pointers large, copying it is very cheap.

Reply:
ok

Reply:
FWIW, C++ has (Named) Return Value Optimization to often remove these kinds of copies automatically by the compiler.